src/CommandLine/CommandLineParser.cpp

```diff
@@ -1150,41 +1161,39 @@ bool CommandLineParser::parseCommandLine(int argc, const char** argv) {
       } else {
         m_sverilog = true;
       }
-    } else if (!all_arguments[i].empty() && all_arguments[i] == "--x-assign") {
+    } else if (all_arguments[i] == "--x-assign") {
       Location loc(mutableSymbolTable()->registerSymbol(all_arguments[i]));
       Error err(ErrorDefinition::CMD_PLUS_ARG_IGNORED, loc);
       m_errors->addError(err);
       i++;
-    } else if (!all_arguments[i].empty() && all_arguments[i] == "--x-initial") {
+    } else if (all_arguments[i] == "--x-initial") {
       Location loc(mutableSymbolTable()->registerSymbol(all_arguments[i]));
       Error err(ErrorDefinition::CMD_PLUS_ARG_IGNORED, loc);
       m_errors->addError(err);
       i++;
-    } else if (!all_arguments[i].empty() && all_arguments[i].at(0) == '+') {
+    } else if (all_arguments[i].at(0) == '+') {
```

Comment:
so here, the removed empty() check is probably still appropriate as we access a particular element in the string.
Or has it been ensured at this point that we can't have an empty arg ?

Reply:
Line 520 and then again on line 676.